addons/crm/models/crm_lead.py

```diff
@@ -1222,13 +1222,10 @@ def _merge_opportunity_history(self, opportunities):
         self.ensure_one()
         for opportunity in opportunities:
             for message in opportunity.message_ids:
+                subject = _("From %(source_name)s", source_name=opportunity.name)
                 message.write({
                     'res_id': self.id,
-                    'subject': _(
-                        "From %(source_name)s : %(source_subject)s",
-                        source_name=opportunity.name,
-                        source_subject=message.subject
-                    )
+                    'subject': message.subject and subject + _(" : %(source_subject)s", source_subject=message.subject) or subject,
```

Comment:
This kind of split sentence is very hard for translators to handle (it's actually impossible to translate that in some languages such as Japanese), I think we should just make a condition above, and have 2 full versions.

Something like that in pseudo code (I didn't check for typos)
```
if message.subject:
    subject = _("From %(source_name)s : %(source_subject)s", source_name=opportunity.name, source_subject=message.subject)
else:
    subject = _("From %(source_name)s", source_name=opportunity.name)
```